### DISCUSSION TITLE
ISO 20022 XSDs

### DISCUSSION THREAD

**Discussion by jhagai** [prowide/prowide-iso20022#14] [Q&A] [answered]
Hello, 

I see this library includes lots of generated code. 
However, I cannot figure out how this code was generated. 
I cannot find any reference to XSDs nor any gradle jaxb plugin (I am more of a maven guy so I might have missed it).

By the way, I suppose those classes are for CBPR+ formats. 
Do you know if they can be used to parse RTGS specific formats (like Target). 
In other words are CBPR+ formats comptible with RTGS (I guess the opposite isn't true).

Thank you,

Joël

**Comment by zubri** [comment, marked as answer]
Hi, yes. Besides the core module providing the glue code and general API, the actual message model is all generated using JAXB. 

The generation process is not published as part of the open source since it uses several proprietary scripts and JAXB plugins. 
It is not a plain generation from the schemas, but a mix, using JAXB as the raw generation and several post-processes. These additional processes are used, for example, to build a common dictionary of POJOS across the library, to reduce duplication. Since JAXB by default would generate thousands or duplicate classes in different packages.

Then regarding the XSDs, the model is generated for the ISO 20022 schemas. These schemas are the general framework, meaning, the generated model is compatible with any restricted "sub-standard" such as CBPR+, T2, and any other ISO 20022-based RGTS.

You can take for instance the model for pacs.008.001.08 and build a message compatible with T2. Just make sure you satisfy the specific schema and rules restrictions.

**Reply by jhagai** [reply]
Thank you very much.
